renderdoc/driver/vulkan/vk_replay.cpp

```diff
@@ -1490,32 +1495,39 @@ void VulkanReplay::SavePipelineState(uint32_t eventId)
     }
 
     // Color Blend
-    ret.colorBlend.alphaToCoverageEnable = p.alphaToCoverageEnable;
-    ret.colorBlend.alphaToOneEnable = p.alphaToOneEnable;
+    ret.colorBlend.alphaToCoverageEnable = state.alphaToCoverageEnable != VK_FALSE;
+    ret.colorBlend.alphaToOneEnable = state.alphaToOneEnable != VK_FALSE;
 
     ret.colorBlend.blends.resize(p.attachments.size());
     for(size_t i = 0; i < p.attachments.size(); i++)
     {
-      ret.colorBlend.blends[i].enabled = p.attachments[i].blendEnable;
+      if(i < state.colorBlendEnable.size())
+        ret.colorBlend.blends[i].enabled = state.colorBlendEnable[i] != VK_FALSE;
```

Comment:
Similar to the earlier comment, this should reset the values to a default (in this case false, similar for other values) if it's not specified.

I also had to check the VUs and I'm not 100% sure - can you clarify what the behaviour should be regarding these counts matching? In most cases for vulkan the counts are required to match when specified in multiple different places, but the VU for this one seems to say you only need to specify a value for "active color attachments". Does that mean if there are attachments which are unused this check may be necessary and it would be valid for the application to have only specified some smaller number of values in these color blend arrays?

Reply:
The changes here now worry me - as they do the opposite from what they did before by assuming that the arrays are all matching in size. I couldn't find any VU actually requiring that which is why I was looking for clarification in the comment above.

Checking again, I can also see this:

>`attachmentCount` is the number of `VkPipelineColorBlendAttachmentState` elements in `pAttachments`. It is ignored if the pipeline is created with VK_DYNAMIC_STATE_COLOR_BLEND_ENABLE_EXT, VK_DYNAMIC_STATE_COLOR_BLEND_EQUATION_EXT, and VK_DYNAMIC_STATE_COLOR_WRITE_MASK_EXT dynamic states set, and either `VK_DYNAMIC_STATE_COLOR_BLEND_ADVANCED_EXT` set or `advancedBlendCoherentOperations` is not enabled on the device.

Which seems like it's possible that `p.attachments` could be entirely empty even though the pipeline does have attachments which it expects to write to (and possibly blend to) depending on dynamic state.

This definitely needs a solid understanding of where these counts are intended to match and where they can be different, as it is right now this doesn't look correct to me.